### PR TITLE
scorch APIs to support rollback

### PR DESCRIPTION
**Pull request by abhinavdangeti**
- PreviousPersistedSnapshot
- SnapshotRevert

+ unit test

### CODE REVIEW

File: index/scorch/introducer.go [hunk 1]
```diff
@@ -241,3 +252,48 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 	// notify merger we incorporated this
 	close(nextMerge.notify)
 }
+
+func (s *Scorch) revertToSnapshot(revertTo *snapshotReversion) error {
+	defer close(revertTo.applied)
+
+	if revertTo.snapshot == nil {
+		return fmt.Errorf("Cannot revert to a nil snapshot")
```
Comment (steveyen) [inline index/scorch/introducer.go]: need to also send this error onto the revertTo.applied channel?

File: index/scorch/snapshot_rollback.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			}
+
+			indexSnapshot.epoch = snapshotEpoch
+			return indexSnapshot, nil
+		}
+
+		return nil, nil
+	}
+
+	return getFromBolt()
```
Comment (steveyen) [inline index/scorch/snapshot_rollback.go]: looks like a helper getFromBolt() func isn't really needed here?  might as well just make it a straight set of code?

File: index/scorch/introducer.go (hunk 1 above)
Comment (abhinavdangeti) [reply index/scorch/introducer.go]: Done.

File: index/scorch/snapshot_rollback.go (hunk 2 above)
Comment (abhinavdangeti) [reply index/scorch/snapshot_rollback.go]: Done.

**Pull request closed without merging**